### ISSUE TITLE
How does one use this to train?

### ISSUE THREAD

**Issue by atorsvn** [oobabooga/text-generation-webui#1168]
Im not clear on the file format format for training? Is it JSON? Text? Is there an example for using this with huggingface datasets?

**Comment by mcmonkey4eva** [comment]
You can use either JSON or simple text files. There's a training tab in the webui and all the inputs are explained.
If you use JSON datasets, you need to provide a format file, and there's an example one (for the Alpaca dataset <https://github.com/gururise/AlpacaDataCleaned/blob/main/alpaca_data_cleaned.json>) provided in the formats folder.
If you use a raw text file, you don't need to worry about that.

**Comment by elcotek** [comment]
I had already downloaded johnson templates from huggingface.co for txt to txt to test, but I don't understand the exact procedure for training. There was always a message for me.. error. /

**Comment by stellarxxu** [comment]
I also download the templates, but with the same error after click button ""Start LoRA Training""

**Comment by cibernicola** [comment]
I'm amble to do "things" with raw (short) texts, but What I guet is almost nothing :( 
https://github.com/oobabooga/text-generation-webui/discussions/1199#discussioncomment-5622935

Example: Raw text (copy paste) from here: https://en.wikipedia.org/wiki/Louis_H._Bean
Training ok (any warning)
This i what I get;
![image](https://user-images.githubusercontent.com/4579387/232233606-eb0e3f19-0be2-4632-b506-719d06492688.png)

content of json file: 
```
{
  "base_model_name_or_path": "models\\decapoda-research_llama-7b-hf",
  "bias": "none",
  "fan_in_fan_out": false,
  "inference_mode": true,
  "init_lora_weights": true,
  "lora_alpha": 512,
  "lora_dropout": 0.05,
  "modules_to_save": null,
  "peft_type": "LORA",
  "r": 256,
  "target_modules": [
    "q_proj",
    "v_proj"
  ],
  "task_type": "CAUSAL_LM"
}
```

content of bin file (brute content):
 
![image](https://user-images.githubusercontent.com/4579387/232233569-016d38fd-0706-4892-95ba-76f27cc26f8b.png)


:/

**Comment by mcmonkey4eva** [comment]
See conversation @ https://github.com/oobabooga/text-generation-webui/pull/1098#issuecomment-1509713861
tldr: looks like current version of peft is bugged with file saving. Using `pin install peft==0.2.0` to go back might fix it.
Alternately, if you use latest peft and that PR, set save steps to 1 and you can possibly grab `pytorch_model.bin` out of the last checkpoint folder and replace `adapter_model.bin` with it? I haven't tested but that might be a functional workaround for the saving bug.
